.github/workflows/typing.yml

```diff
@@ -0,0 +1,40 @@
+name: Type checking
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  mypy:
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: "true"
+          cache-suffix: ${{ matrix.python-version }}
+          cache-dependency-glob: "pyproject.toml"
+      - name: Create venv
+        run: uv venv .venv
+      - name: install-reqs
+        # TODO: add more dependencies/backends incrementally
+        run: |
+          source .venv/bin/activate
+          uv pip install -e ".[dev]" polars
```

Comment:
Ok if we just have polars as dependency, it's all green.

It is in line with what we have in `main` actually, given that the only 2 `additional-dependencies` are `pytest` and `polars`.

It is a bit too early to merge IMO, given that `pre-commit` will stop any commits from environments that have anything more than polars. I'll try to fix or ignore all errors that comes when `core` is installed.

🚧 _work in progress_ 🚧

Reply:
upsi @MarcoGorelli you were a faster typer. :D What do you think?

Reply:
not sure what you mean by

>  pre-commit will stop any commits from environments that have anything more than polars.

could you elaborate pleas?

Reply:
Given that `mypy` is running in `pre-commit` in `local` "mode", it will check the repo using every dependency one has installed locally.

At the moment, the checks will pass only if you install the dev dependencies with `uv pip install -e ".[dev]" polars`

Already adding an extra dependency (e.g. `pyarrow`) will make `mypy` fail -> `pre-commit` complain

I think we should merge this only after `mypy` passes with at least all the libraries in `core`

Reply:
ah i see, thanks!

can't we ignore the missing imports from pyarrow in pyproject.toml? like

https://github.com/pola-rs/polars/blob/ff994f0502bc5ce886df43ffbbf468a79adcf281/py-polars/pyproject.toml#L132-L137

Reply:
That's helping reducing a bit the errors, I also disabled `strict = true`

I tried my best to tell `mypy` to calm down a bit, but it still shows a lot of errors. Does anyone know any trick to tell mypy to ignore errors?
I tried [this](https://mypy.readthedocs.io/en/stable/config_file.html#confval-ignore_errors) but didn't have any effect
 
I guess the alternative is to fix them here. 🥲